dags/atd-parking-data.py

```diff
@@ -0,0 +1,263 @@
+    "passport_DB",
+    "smartfolio_s3",
+    "match_field_processing",
+    "parking_socrata_dataset_payments",
+    "parking_socrata_dataset_fiserv",
+    "parking_socrata_dataset_transactions",
+]
+
+
+@task
+def decide_prev_month(prev_exec):
+    """
+    Determines if the current month or the current plus previous month S3
+        folders are needed. If it is within a week of the previous month,
+        also upsert that months data.
+    Parameters
+    ----------
+    prev_execution_date_success : String
+        Last date the flow was successful.
+
+    Returns
+    -------
+    Prev_month : Bool
+        Argument if the previous month should be run.
+
+    """
+    last_date = datetime.strptime(prev_exec, "%Y-%m-%d")
+    # If in the first 8 days of the month of last few days of the month re-run
+    # the previous month's data to make sure it is complete.
+    if last_date.day < 8 or last_date.day > 26:
+        prev_month = True
+    else:
+        prev_month = False
+
+
+@task
+def add_command_arguments(commands, prev_exec, prev_month):
+    # Adding parameters to to our docker commands to include the previous exec date
+    # and wether or not to run the previous month
+
+    output_tasks = []
+    for c in commands:
+        if "txn_history.py" in c:
+            c = f"{c} --start {prev_exec}"
+
+        if "passport_txns.py" in c:
+            c = f"{c} --start {prev_exec}"
+
+        if "fiserv_DB.py" in c:
+            c = f"{c} --lastmonth {prev_month}"
+
+        if "payments_s3.py" in c:
+            c = f"{c} --lastmonth {prev_month}"
+
+        if "passport_DB.py" in c:
+            c = f"{c} --lastmonth {prev_month}"
+
+        if "smartfolio_s3.py" in c:
+            c = f"{c} --lastmonth {prev_month}"
+
+        output_tasks.append(c)
+    return output_tasks
+
+
+with DAG(
+    dag_id="atd_parking_data",
+    description="Scripts that download and process parking data for finance reporting.",
+    default_args=default_args,
+    schedule_interval="35 8 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    dagrun_timeout=timedelta(minutes=60 * 4),
+    tags=["repo:atd-parking-data", "parking", "socrata", "postgrest"],
+    catchup=False,
+) as dag:
+    env_vars = get_env_vars_task(REQUIRED_SECRETS)
+    prev_exec = "{{ (prev_start_date_success - macros.timedelta(days=7)).strftime('%Y-%m-%d') if prev_start_date_success else '2023-08-01'}}"
+
+    prev_month = decide_prev_month(prev_exec)
+
+    tasks = add_command_arguments(docker_commands, prev_exec, prev_month)
+
+    docker_tasks = []
+    for i, docker_command in enumerate(docker_commands):
+        docker_task = DockerOperator(
+            task_id=task_ids[i],
+            image=docker_image,
+            command=docker_command,
+            api_version="auto",
+            auto_remove=True,
+            environment=env_vars,
+            dag=dag,
+            tty=True,
+            force_pull=False,
+        )
+        docker_tasks.append(docker_task)
```

Comment:
meat of what's going on, we're creating a list of `DockerOperator`s and then running them sequentially.